scripts/extension_helpers.js

```diff
@@ -56,58 +61,58 @@ VENDOR_FILES = VENDOR_FILES.filter(function (importPath) {
 */
 
 
-// The base externals for JupyterLab itself
-
-
 /**
- * A function that mangles phosphor imports.
+ * The Webpack `externals` config for JupyterLab itself.
  */
-function phosphorExternals(context, request, callback) {
+var BASE_EXTERNALS = [
+  function(context, request, callback) {
     // All phosphor imports get mangled to use the external bundle.
-    var regex = /^phosphor\/lib\/([\w\/]+)$/;
+    var regex = /^phosphor\/lib\/[\w\/]+$/;
     if(regex.test(request)) {
-        var matches = regex.exec(request)[1];
-        var lib = 'var phosphor.' + matches.split('/').join('.');
+        var path = require.resolve(request).replace('.js', '');
+        var index = path.indexOf('phosphor/lib');
+        path = path.slice(index + 'phosphor/lib/'.length);
+        var lib = 'var jupyter.phosphor.' + path.split('/').join('.');
         return callback(null, lib);
     }
     callback();
-}
-
-
-var BASE_EXTERNALS = [
-  phosphorExternals,
+  },
   {
-    'jupyter-js-services': 'jupyter.services',
     'jquery': '$',
     'jquery-ui': '$'
   }
 ];
 
 
-// Downstream extensions should exclude JupyterLab itself as well.
-var DEFAULT_EXTERNALS = BASE_EXTERNALS + [
-    function(context, request, callback) {
-      // JupyterLab imports get mangled to use the external bundle.
-      var regex = /^jupyterlab\/lib\/([\w\/]+)$/;
-      if(regex.test(request)) {
-          var matches = regex.exec(request)[1];
-          var lib = 'var jupyterlab.' + matches.split('/').join('.');
-          return callback(null, lib);
+/**
+ * The default Webpack `externals` config that should be applied to
+ * extensions of JupyterLab.
+ */
+var EXTENSION_EXTERNALS = BASE_EXTERNALS.concat([
+  { 'jupyter-js-services': 'jupyter.services' },
+  function(context, request, callback) {
+    // JupyterLab imports get mangled to use the external bundle.
+    var regex = /^jupyterlab\/lib\/([\w\.\/]+)$/;
+    if(regex.test(request)) {
+      try {
+        var path = require.resolve(request).replace('.js', '');
```

Comment:
Actually, it's probably simpler to just store the `.js` with the filename, so you don't have to strip it out both here and when you store it.